phoenix-core-client/src/main/java/org/apache/phoenix/expression/function/DecodeBinaryFunction.java

```diff
@@ -0,0 +1,138 @@
+package org.apache.phoenix.expression.function;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.Base64;
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.exception.SQLExceptionInfo;
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.parse.FunctionParseNode;
+import org.apache.phoenix.schema.IllegalDataException;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PVarchar;
+import org.apache.phoenix.schema.types.PVarbinary;
+import org.apache.phoenix.schema.tuple.Tuple;
+
+/**
```

Comment:
You have the names mixed up.

Encoding is BINARY -> STRING

Decoding is STRING -> BINARY

See the semantics in DecodeFunction.